deep_trekker.orogen

```diff
@@ -1,55 +1,69 @@
-# these projects should be imported there as well.
-import_types_from "std"
-# import_types_from "base"
-
-# Declare a new task context (i.e., a component)
-#
-# The corresponding C++ class can be edited in tasks/Task.hpp and
-# tasks/Task.cpp, and will be put in the deep_trekker namespace.
-task_context "Task" do
-    # This is the default from now on, and should not be removed. Rock will
-    # transition to a setup where all components use a configuration step.
+import_types_from "deep_trekker/DeepTrekkerCommands.hpp"
+import_types_from "deep_trekker/DeepTrekkerStates.hpp"
+import_types_from "iodrivers_base"
+import_types_from "base"
+
+task_context "RevolutionTask" do
     needs_configuration
 
-    # A configuration property (here, a std::string). Its value can be retrieved
-    # in the C++ code with # _config_value.get() and _config_value.set(new_value).
-    property "config_value", "/std/string"
-
-    # An input port, i.e. an object from which the component gets data from
-    # other components' outputs
-    #
-    # Data can be retrieved using _input.read(value), which returns true if data
-    # was available, and false otherwise. _input.connected() returns if this
-    # input is connected to an output or not.
-    input_port "input", "/std/string"
-
-    # An output port, i.e. an object to which the component pushes data so that
-    # it is transmitted to other components' inputs
-    #
-    # Data can be written using _output.write(value). _output.connected() returns
-    # if this output is connected to an input or not.
-    output_port "output", "int"
-
-    # If you want that component's updateHook() to be executed when the "input"
-    # port gets data, uncomment this and comment the 'periodic' line
-    # port_driven "input"
-    # By default, the task will be periodic with a period of 0.1
-    periodic 0.1
-end
+    # API version
+    property "api_version", "std/string"
+
+    # Devices mac adress
+    property "devices_mac_address", "deep_trekker/DevicesMacAddress"
+
+    # Tilt camera head position/speed joint command
+    input_port "tilt_camera_head_command", "base/samples/Joints"
+
+    # Camera head laser, light and camera commands
+    input_port "camera_head_command", "deep_trekker/CameraHeadCommand"
+
+    # Grabber raw joint commands
+    input_port "grabber_command", "base/samples/Joints"
+
+    # Vehicle setpoint in local frame and light command
+    input_port "rov2ref_command", "deep_trekker/PositionAndLightCommand"
+
+    # Powered reel speed joint command
+    input_port "powered_reel_command", "base/samples/Joints"
+
+    # Rawdata in
+    input_port "data_in", "iodrivers_base/RawPacket"
+
+    # Rawdata out
+    output_port "data_out", "iodrivers_base/RawPacket"
 
+    # Rawdata vehicle command out (debug purpose)
+    output_port "data_command_out", "iodrivers_base/RawPacket"
+
+    # Rawdata query device info out (debug purpose)
+    output_port "data_query_out", "iodrivers_base/RawPacket"
```

Comment:
We should be able to find these in `data_out` directly.

Reply:
How can I do that in the test? Because always that we pass trough the update, we query data and sends data (or not) using the same output....so, what happens is that, always in the test I get just the output from the queried command.

Reply:
You can filter the samples received with `.matching`

```
have_one_new_sample(port).matching do |s|
   json = JSON.parse(s.data.to_byte_array[8...-1])
   // some criteria on json that allows to find the right packet, e.g.
   json['type'] == "GET"
end
```

Reply:
nice!